### ISSUE TITLE
Llvmir2hll does not remove "__decompiler_undefined_function_0"

### ISSUE THREAD

**Issue by crowman2** [avast/retdec#750]
Hello.
I have a very simple binary "part.bin": [part.zip](https://github.com/avast/retdec/files/4516993/part.zip)

Using decompiler with options:
`--select-ranges 0x0-0x2E --select-decode-only -m raw -a x86 -e little --raw-entry-point 0x0 --raw-section-vma 0x0`

As the result I've got correct C source, but wrong IR with two "decompiler_undefined_functions":
```
source_filename = "test"
target datalayout = "e-p:32:32-f64:32:64-f80:32-n8:16:32-S128"

define i32 @entry_point() local_unnamed_addr {
dec_label_pc_0:
  %storemerge.reg2mem = alloca i32, !insn.addr !0
  %0 = call i32 @__decompiler_undefined_function_0()
  %1 = call i1 @__decompiler_undefined_function_1()
  %stack_var_-12 = alloca i32, align 4
  %2 = icmp eq i1 %1, false, !insn.addr !1
  store i32 4, i32* %storemerge.reg2mem, !insn.addr !1
  br i1 %2, label %dec_label_pc_23, label %dec_label_pc_2c, !insn.addr !1

dec_label_pc_23:                                  ; preds = %dec_label_pc_0
  %esp.0 = ptrtoint i32* %stack_var_-12 to i32
  %3 = add i32 %esp.0, -4, !insn.addr !2
  %4 = inttoptr i32 %3 to i32*, !insn.addr !2
  store i32 0, i32* %4, align 4, !insn.addr !2
  store i32 0, i32* %storemerge.reg2mem, !insn.addr !3
  br label %dec_label_pc_2c, !insn.addr !3

dec_label_pc_2c:                                  ; preds = %dec_label_pc_0, %dec_label_pc_23
  %storemerge.reload = load i32, i32* %storemerge.reg2mem
  store i32 %storemerge.reload, i32* inttoptr (i32 4202496 to i32*), align 8192
  ret i32 %0, !insn.addr !4
}

declare i32 @__decompiler_undefined_function_0() local_unnamed_addr
declare i1 @__decompiler_undefined_function_1() local_unnamed_addr

!0 = !{i64 0}
!1 = !{i64 4}
!2 = !{i64 37}
!3 = !{i64 38}
!4 = !{i64 45}
```
It seems during "Running phase: removing functions prefixed with [__decompiler_undefined_function_]" decompiler did not delete them correctly.
Is it a bug?

**Comment by PeterMatula** [comment]
Related to #301, both should be fixed once branch `retdec-lib` is merged to `master`.

**Comment by PeterMatula** [comment]
Fixed in bcf5600 in `retdec-lib`. The issue gets closed automatically when merged to `master`.
